ocean_lib/services/v4/service.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""
+    Service Class for V4
+    To handle service items in a DDO record
+"""
+import copy
+import logging
+from typing import Dict, Any, Optional
+
+from enforce_typing import enforce_types
+
+from ocean_lib.common.agreements.service_types import ServiceTypesV4, ServiceTypesNames
+
+logger = logging.getLogger(__name__)
+
+
+@enforce_types
+class NFTService:
+    """Service class to create validate service in a V4 DDO."""
+
+    SERVICE_TYPE = "type"
+    SERVICE_FILES = "files"
+    SERVICE_NAME = "name"
+    SERVICE_DESCRIPTION = "description"
+    SERVICE_DATATOKEN = "datatokenAddress"
+    SERVICE_ENDPOINT = "serviceEndpoint"
+    SERVICE_TIMEOUT = "timeout"
+
+    def __init__(
+        self,
+        service_type: str,
+        service_endpoint: str,
+        data_token: str,
+        files: str,
+        timeout: float,
+        other_values: Optional[Dict[str, Any]] = None,
```

Comment:
do we still allow other_values? Either way, I'd reorder the args so that other_values comes last.

Reply:
I was thinking to keep them for the compute dictionary. But now I think that is suitable to have them just for the service class right?

Reply:
I'm not sure. I would discuss this with @alexcos20 or @trentmc, should we support extra keys in the DDO?